### DISCUSSION TITLE
Can this package be used to add roles/ permissions for a Passport service-to-service client?

### DISCUSSION THREAD

**Discussion by SuperDJ** [spatie/laravel-permission#2429] [Q&A]
Currently this package seems to only support user based authentication. This also works fine with most of the Passport client types as long as a user authenticates. However there is also a service-to-service Passport client. In that case the roles/ permissions, I think, should be associated with the client. Is this possible or does this require a PR?

**Comment by parallels999** [comment]
Feel free to make a PR, and also add tests for that

**Reply by SuperDJ** [reply]
@parallels999 do you have any suggestion of how to do this. Because right now I run into issues with the client not being associated with a user. The issues are related to how Laravel handles authentication with guards and providers.

**Comment by SuperDJ** [comment]
If someone is interested, here is a start https://github.com/spatie/laravel-permission/pull/2467.
